paywall/src/data-iframe/blockchainHandler/purchaseKey.js

```diff
@@ -0,0 +1,60 @@
+import { getAccount } from './account'
+import ensureWalletReady from './ensureWalletReady'
+import submittedListener from './purchaseKey/submittedListener'
+import updateListener from './purchaseKey/updateListener'
+
+export async function purchaseKey({
+  walletService,
+  window,
+  lockAddress,
+  amountToSend,
+}) {
+  await ensureWalletReady(window)
+  const account = getAccount()
+
+  return walletService.purchaseKey(lockAddress, account, amountToSend)
+}
+
+export async function processKeyPurchaseTransactions({
```

Comment:
Do you mind adding the comment you put n the PR here please: 
> If the page is refreshed in the middle of the process, processKeyPurchaseTransactions is re-entrant, and will pick up where it left off.

Reply:
sure. I also have a commit I forgot to push that has full jsdoc comments, sorry about not including that earlier